.github/workflows/pre-integration.yml

```diff
@@ -21,11 +21,12 @@ jobs:
     name: Test JDK ${{ matrix.java }}, ${{ matrix.os }}, ${{ matrix.profile }}
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up JDK ${{ matrix.java }}
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v3
       with:
         java-version: ${{ matrix.java }}
+        distribution: 'temurin'
     - name: Cache Maven packages
       uses: actions/cache@v2
```

Comment:
Now you could replace `actions/cache@v2` usage entirely by [specifying the input `cache: 'maven'` to `actions/setup-java@v3`](https://github.com/actions/setup-java#caching-maven-dependencies).